cogs/respects.py

```diff
@@ -161,7 +161,10 @@ def __init__(self, bot):
                                                 before=ctx.message):
                 prevMsgs.append(msg.id)
 
-            exceedMessages = self.settings[sid][cid][KEY_MSG].id not in prevMsgs
+            if self.settings[sid][cid][KEY_MSG]:

```

Comment:
So... If the bot doesn't have send message perms this would be empty?

Reply:
Yeah, on (line 229)[https://github.com/SFUAnime/Ren/blob/V2/prod/cogs/respects.py#L229], it'll error out on the previous line before the assignment.

Reply:
Then, I'm not sure how this resolves the issue about failing gracefully if there are no perms... I guess this fixes an issue if there are no previous respects?

Reply:
Sorry, should have been more specific.  This fixes the case where someone types `renf` in a channel where send permissions are denied (e.g. #dark-hour).  When this happens, the message object is not saved to `...[KEY_MSG]` as it errors out in line 228 mentioned above.  At that point, the sid and cid of the server and channel IDs have been set in the dict, respectively.  This fixes the case where if send message perms are re-enabled in the channel, the check above doesn't error out due to accessing a non-existent Message object, since we already had a previous respect being paid (albeit didn't send properly).

Reply:
Oh. I see. Wow. This is quite the edge case....

Reply:
~~This info should've been in that issue you put out if that's what this aims to fix...~~

Reply:
~~Let me put it in after the fact 😂 ~~